src/libpmemobj/palloc.c

```diff
@@ -838,6 +838,30 @@ palloc_pointer_compare(const void *lhs, const void *rhs)
 	return 0;
 }
 
+VEC(pobj_actions, struct pobj_action);
+
+/*
+ * pobj_actions_add -- add new action to the end of the vector and return
+ *      its slot. Vector must be able to hold the new value. Reallocation is
+ *      forbidden.
+ */
+static struct pobj_action *
+pobj_actions_add(struct pobj_actions *actv)
```

Comment:
This is effectively just a debug check, so I'm not sure if it's worth adding all these runtime error handling.
So, I'd just simplify this to `ASSERTne(size, capacity); VEC_PUSH_BACK(actv);`

Reply:
That's the whole point of this change - to catch broken code ASAP and not corrupt user data in the release builds, where asserts are compiled out.

Reply:
I mean, that's the whole point of ASSERTs. By that logic, we should leave them enabled in release builds just in case?
I don't know, it just seems odd to me.

Reply:
I removed error handling and added abort() when the error condition is triggered.